meson.build

```diff
@@ -15,6 +15,50 @@ endif
 libzim_dep = dependency('libzim', version : '>=6.1.1', static:static_linkage)
 
 compiler = meson.get_compiler('cpp')
+find_library_in_compiler = meson.version().version_compare('>=0.31.0')
 rt_dep = compiler.find_library('rt', required:false)
 
+with_writer = target_machine.system() != 'windows'
+
+if with_writer
+  thread_dep = dependency('threads')
+  libzim_dep = dependency('libzim', version : '>=6.1.2', static:static_linkage)
+  zlib_dep = dependency('zlib', static:static_linkage)
```

Comment:
zlib is used by `tools.cpp` and it is in the "global" src directory.
It is indeed compiled only if we compile zimwriterfs (everywhere but windows) but as it is in the global src, it will ends to be compiled for other binaries. As zlib is a dependency of libzim, it will be always available. We can move this test "globally".

In fact, this is the same for gumbo and magic. Maybe moving the tools.cpp in the global directory was not a good idea :/

Reply:
Actually from the entire tools.(h|cpp)  libmagic is used by `getMimeTypeForFile()`   and zlib is used by `inflateString()`
We can move those two function into src/zimwriterfs/tools.cpp, leaving the rest in src/tools.cpp

Reply:
@mgautierfr What you think about splitting tools.cpp ?

Reply:
removed the duplicate of libzim_dep; splitted tools.(h|cpp) to src/tools.(h|cpp) and src/zimwriterfs/tools.(h|cpp) with functions which depends on zlib, magic and gumbo.